doc/api/test.md

```diff
@@ -219,6 +219,46 @@ test('a test that creates asynchronous activity', (t) => {
 });
 ```
 
+## Running tests from the command line
+
+The Node.js test runner can be invoked from the command line by passing the
+[`--test`][] flag. By default, Node.js will recursively search the current
+directory for test files:
+
+```bash
+node --test
+```
+
+Alternatively, one or more paths can be provided as the final argument(s) to
+the Node.js command, as shown below.
+
+```bash
+node --test test1.js test2.mjs test/
+```
+
+In this example, the test runner will execute the files `test1.js` and
+`test2.mjs`. The test runner will also recursively search the `test/` directory
+for test files to execute.
+
+When searching for test files to execute, the test runner behaves as follows:
```

Comment:
these filename conventions are very very arbitrary, and don't actually match some of the most common patterns - a top-level `test` directory inside which all files are tests, and no files outside it are.

Reply:
They aren't totally arbitrary. I looked at what the core test suite uses as well as some other runners. That said, if you think adding `test/` is important, that can be discussed.

Reply:
node's own repo is almost entirely different from what the actual JS ecosystem does, including its linting and testing strategies.

A top-level `test` dir was the prevailing pattern in the community by a *large* margin prior to jest popularizing colocated tests, and is still heavily used.

Reply:
I'm seeing the same patterns as @ljharb is describing, either a top-level `test` directory (wherein not all files have to be tests, but utils simply don't do anything when called directly), or the colocated tests, with `.test.js` suffix.

Here's the jest naming pattern, for reference: https://jestjs.io/docs/configuration#testmatch-arraystring

Reply:
I'm fine with supporting a top-level `test` directory. That's how I handle tests in all of my own repos anyway.

> wherein not all files have to be tests, but utils simply don't do anything when called directly

I'm not sure if I'm understanding you correctly, but I don't think we should classify a file as a test or not based on what happens when you run it. For example, say you have `test/utils.js` that only provides helpers for your test suite. The test runner should execute the file - presumably it won't output any TAP but will exit with code 0. IMO, the test runner should still treat that file as a passing test. Otherwise, we will restrict ourselves only to files that output TAP.

Reply:
but think about the coverage and stat, `test/util.js` will cause the test result count +1.

and what if `test/futures/`?